core/headerchain.go

```diff
@@ -92,16 +92,49 @@ func NewHeaderChain(db ethdb.Database, engine consensus.Engine, chainConfig *par
 	return hc, nil
 }
 
+func (hc *HeaderChain) ManifestForHeader(h *types.Header) (types.BlockManifest, error) {
+	manifest := types.BlockManifest{}
+	if h.NumberU64() == 0 {
+		return manifest, nil
+	}
+	for {
+		ancestor := hc.GetHeader(h.ParentHash(), h.NumberU64()-1)
+		if ancestor == nil {
+			return types.BlockManifest{}, errors.New("ancestor not found")
+		}
+		ancHash := ancestor.Hash()
+		manifest = append(manifest, &ancHash)
```

Comment:
It might be more clear if we use the actual values since if people don't use the accessor methods, they will end up using the wrong value though this approach is definitely more efficient in terms of memory use.

Reply:
For now I think we should leave manifest as an array of pointers to hashes, since that is consistent with all the other types in `block.go`, e.g:
```
type Transactions []*Transaction

type extblock struct {
	Header      *Header
	Txs         []*Transaction
	Uncles      []*Header
	Etxs        []*Transaction
	SubManifest []*common.Hash
}
```

Reply:
Yes, just have to keep in mind while using the manifest. We should enforce the use of acessors.